### PR TITLE
Fix/carthage integration tests

### PR DESCRIPTION
**Pull request by aboedo**
Our carthage integration tests have a couple of issues: 
- They have a reference to the local project as a git reference, so carthage actually git clones it into its own folder, and uses the default branch. I fixed it by adding an extra step to update the git pointer, very similar to what we did for SPM integration tests. 
- Carthage recursively inspects all folders in search for shared schemes in xcprojects. This means that unfortunately, when doing `carthage build`, it tries to build all other integration tests. And the SPM integration tests **within** the carthage integration tests fail to check out _their_ dependencies, and I'm not 100% sure why that is, since the ssh keys were added in a previous step. But in any case, we don't want to build the other integration tests. 
So I fixed it by doing: 
`carthage update --no-build` - this does the clone
`rm -rf integration tests` - removes all integration tests from the clone, also preventing infinite recursion
`carthage build` from the cloned project

I'm fine, why do you ask?

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -269,11 +283,12 @@ workflows:
...
   build-test:
     jobs:
-      - runtest
+      # - runtest
       - deployment-checks: *release-branches
       - integration-tests-cocoapods: *release-tags-and-branches
       - integration-tests-swift-package-manager: *release-tags-and-branches
-      - integration-tests-carthage: *release-tags-and-branches
+      - integration-tests-carthage
+      # - integration-tests-carthage: *release-tags-and-branches
```
Comment (aboedo) [inline .circleci/config.yml:291]: will revert these changes before merging

File: IntegrationTests/CarthageIntegration/fastlane/Fastfile [hunk 2]
```diff
@@ -1 +1,9 @@
-import("../../../fastlane/Fastfile")
\ No newline at end of file
+import("../../../fastlane/Fastfile")
+
+  desc "Update carthage commit"
```
Comment (vegaro) [inline IntegrationTests/CarthageIntegration/fastlane/Fastfile:3]: The indentation here looks off

File: .circleci/config.yml [hunk 3]
```diff
@@ -237,15 +246,20 @@ jobs:
...
             - carthage-cache-{{ checksum "Cartfile.resolved" }}
       - run:
           name: Carthage Update
           working_directory: IntegrationTests/CarthageIntegration/
-          command: ./carthage.sh update --cache-builds
+          # install without building, then remove the tests and build, so that carthage
+          # doesn't try to build the other integration tests
+          command: |
+              ./carthage.sh update --no-build
+              rm -rf Carthage/Checkouts/purchases-root/IntegrationTests/
```
Comment (vegaro) [inline .circleci/config.yml:261]: Isn't this also removing `Carthage/Checkouts/purchases-root/IntegrationTests/CarthageIntegration`?

Comment (aboedo) [reply .circleci/config.yml:272]: yeah, but that's kinda intentional - it's only removing it from the _cloned_ version of the repo, which isn't needed.

File: IntegrationTests/CarthageIntegration/fastlane/Fastfile (hunk 2 above)
Comment (aboedo) [reply IntegrationTests/CarthageIntegration/fastlane/Fastfile]: fixed

File: .circleci/config.yml (hunk 1 above)
Comment (vegaro) [reply .circleci/config.yml:302]: Don't forget to revert 🕺🏼

**Review by vegaro — APPROVED** [review]
